src/rgw/rgw_rados.cc

```diff
@@ -3318,6 +3322,58 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
       filter = &*buffering;
     }
 
+    /*
+     * Presently we don't support ETag based verification if compression or
+     * encryption is requested. We can enable simultaneous support once we have
+     * a mechanism to know the sequence in which the filters must be applied.
```

Comment:
this should work fine with compression. as long as you compose this on top of the compression filter, the etag filter will see the uncompressed data

Reply:
I wasn't sure about the sequence of filter invocation and how to ensure a certain order to it. How can I ensure that etag filter will be invoked only after compression filter?

Reply:
you want it to be invoked _before_ compression so that you're calculating checksums on the uncompressed data

the ordering is determined by the pipelining of these filters; so you want your ETagVerifier's `next` filter to be the compression one. but you've already accomplished this, because you're initializing the ETagVerifier after the compression filter

Reply:
Okay, will remove the check for `!plugin`